gui/renderer.go

```diff
@@ -49,10 +49,29 @@ func (r *OpenGLRenderer) CellHeight() float32 {
 }
 
 func (r *OpenGLRenderer) Clean() {
+	for _, rect := range r.rectangles {
+		rect.Free()
+	}
+
 	r.rectangles = map[[2]uint]*rectangle{}
 }
 
-func (r *OpenGLRenderer) newRectangle(x float32, y float32, colourAttr uint32) *rectangle {
+func (r *OpenGLRenderer) initRectangle(rect *rectangle, x float32, y float32, colourAttr uint32) {
+
+	if rect.vao != 0 {
```

Comment:
Might be worth checking `rect != nil` before checking properties?

Reply:
@liamg have added the check, thanks